### DISCUSSION TITLE
Configuration file question

### DISCUSSION THREAD

**Discussion by woxin5295** [SeismicSource/sourcespec#45] [Q&A]
Dear  Prof. Satriano

1. I have seen the following information transiently in the configuration file, but now I generate the default config file without this information. I need to change the value of kp or ks.
......
......
kp = 0.38
ks = 0.20
# ks = 0.3724

2. Can the code handle stations with similar epicenter distances? because I only see a parameter, "win_length = 5.0" for noise and signal. My station's epicenter distance ranges from 5km to 300km. This single "win_length“ value cannot fit all stations. There are short-period stations as well as broadband stations, and I would like to use them all, otherwise, the azimuth gap is too large for the earthquake.

**Comment by woxin5295** [comment]
Dear Prof. Satriano
I've solved the first question.
But there is a new problem. May you provide an example for ssp_event.yaml.? The code doesn't recognize my input format. Some error messages are given as follows:
.... ....
Reading station metadata: done
---------------------------------------------------
ssp_event.yaml: Not a SourceSpec Event File
Parsing error: This is a valid YAML file, but the following error occurred: __init__() got an unexpected keyword argument 'type'.
ssp_event.yaml: Not a hypo71 hypocenter file
Parsing error: Cannot read origin time on first line.
ssp_event.yaml: Not a hypo2000 hypocenter file
Parsing error: Could not find hypocenter dat

**Comment by claudiodsf** [comment]
This problem is fixed in c4faf0e12f8d3b818e1c158d41774d83732ac5f3.

Thanks for reporting it!

**Reply by krisvanneste** [reply]
@claudiodsf , there is also a second question about a variable (distance-dependent) `win_length` parameter, which I find interesting. Another colleague has asked me this too, and recently I found that 5 s is not enough for stations at > 400 km distance. Do you know some generic function that could be used to define an appropriate window length based on distance?

**Reply by claudiodsf** [reply]
Hi Kris,

Thanks for bringing this up. 

Having a variable `win_length` is in my todo list for the next year (after v1.8 is released). 

However, I still haven't got a clear idea on how to implement it. 
If you want to tackle it, you are more than welcome!

My current solution is to use a long window for close stations: my experience shows that the results are not significantly affected by including signal after the S-wave at these closest stations, since surface waves are not yet dominant and noise is taken into account by the signal-to-noise weighting.

**Reply by krisvanneste** [reply]
Claudio,

Thank you for your answer. The implementation itself should not be too difficult, but I haven't found a function yet. I know of windowing functions that are meant to capture the whole S/Lg window, but that's not the same as the duration required to compute a reliable moment magnitude. Maybe I will have some time in January for this.
